base/distributed/cluster.jl

```diff
@@ -210,13 +210,27 @@ end
 # The master process uses this to connect to the worker and subsequently
 # setup a all-to-all network.
 function read_worker_host_port(io::IO)
-    while true
-        conninfo = readline(io)
+    t0 = time()
+    timeout = worker_timeout()
+
+    ntries = 1000
```

Comment:
We expect the workers to first print out the host:port it is listening on STDOUT.  Ignoring non-matching lines is to make the process of reading host:port a bit more resilient if the spawned command is printing informational/warning messages.

 In the case where the `ssh` command exits with an error, `readline(io)` returns an empty string (`io` being the output stream of the spawned command) and hence we need a limit on the number of tries.

I don't really see a need to make this configurable and documented at this time.